### PR TITLE
Add chocolatey support

### PR DESCRIPTION
**Pull request by Nate-Wilkins**
In accordance with #129

Will require manual changes to the nuspec file on new versions.
I've added an automation script to make it a bit easier. It requires `msbuild`, `gitversion`, and `choco`

### CODE REVIEW

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBusExplorer 4.0.100-PullRequest0140.2.build.267 completed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/build/4.0.100-PullRequest0140.2.build.267) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/d676d344ad by @Nate-Wilkins)

File: build.cmd [hunk 1]
```diff
@@ -0,0 +1,9 @@
+set version=
```
Comment (SeanFeldman) [inline build.cmd]: Needs to work with the automated build

File: src/ServiceBusExplorer/ServiceBusExplorer.nuspec [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Do not remove this test for UTF-8: if “Ω” doesn’t appear as greek uppercase omega letter enclosed in quotation marks, you should use an editor that supports UTF-8, not this one. -->
```
Comment (SeanFeldman) [inline src/ServiceBusExplorer/ServiceBusExplorer.nuspec]: I'm not sure about this tbh

File: src/ServiceBusExplorer/ServiceBusExplorer.nuspec [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+<package>
+  <metadata>
+    <id>servicebusexplorer</id>
+    <title>ServiceBusExplorer</title>
+    <version>4.0.99</version>
+    <authors>paolosalvator</authors>
+    <owners>Nate-Wilkins</owners>
```
Comment (SeanFeldman) [inline src/ServiceBusExplorer/ServiceBusExplorer.nuspec]: Though @paolosalvatori was 😉

Comment (Nate-Wilkins) [inline src/ServiceBusExplorer/ServiceBusExplorer.nuspec]: Oops nice catch 😄

File: src/ServiceBusExplorer/ServiceBusExplorer.nuspec (hunk 2 above)
Comment (Nate-Wilkins) [inline src/ServiceBusExplorer/ServiceBusExplorer.nuspec]: Yeah I can remove this, it's part of the default nuspec given for new choco packages.

File: src/ServiceBusExplorer/ServiceBusExplorer.nuspec [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    <description>The Service Bus Explorer allows users to connect to a Service Bus namespace and administer messaging entities in an easy manner. The tool provides advanced features like import/export functionality or the ability to test topic, queues, subscriptions, relay services, notification hubs and events hubs.</description>
+    <projectUrl>https://github.com/paolosalvatori/ServiceBusExplorer</projectUrl>
+    <projectSourceUrl>https://github.com/paolosalvatori/ServiceBusExplorer</projectSourceUrl>
+    <packageSourceUrl>https://github.com/Nate-Wilkins/chocolatey-packages</packageSourceUrl>
+    <tags>servicebusexplorer azure service bus explorer</tags>
+    <copyright>Copyright 2017 Microsoft</copyright>
+    <licenseUrl>https://github.com/paolosalvatori/ServiceBusExplorer#license</licenseUrl>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <iconUrl>https://raw.githubusercontent.com/Nate-Wilkins/chocolatey-packages/master/icons/servicebusexplorer.png</iconUrl>
```
Comment (tomkerkhove) [inline src/ServiceBusExplorer/ServiceBusExplorer.nuspec]: Maybe this should be moved to the repo as well?

File: src/ServiceBusExplorer/ServiceBusExplorer.nuspec [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    <id>servicebusexplorer</id>
+    <title>ServiceBusExplorer</title>
+    <version>4.0.99</version>
+    <authors>paolosalvator</authors>
+    <owners>Nate-Wilkins</owners>
+    <summary>The Service Bus Explorer allows users to connect to a Service Bus namespace and administer messaging entities in an easy manner. The tool provides advanced features like import/export functionality or the ability to test topic, queues, subscriptions, relay services, notification hubs and events hubs.</summary>
+    <description>The Service Bus Explorer allows users to connect to a Service Bus namespace and administer messaging entities in an easy manner. The tool provides advanced features like import/export functionality or the ability to test topic, queues, subscriptions, relay services, notification hubs and events hubs.</description>
+    <projectUrl>https://github.com/paolosalvatori/ServiceBusExplorer</projectUrl>
+    <projectSourceUrl>https://github.com/paolosalvatori/ServiceBusExplorer</projectSourceUrl>
+    <packageSourceUrl>https://github.com/Nate-Wilkins/chocolatey-packages</packageSourceUrl>
```
Comment (tomkerkhove) [inline src/ServiceBusExplorer/ServiceBusExplorer.nuspec]: Is this intentionally?

**Comment by tomkerkhove** [comment]
Added some "concerns" as well.

**Comment by paolosalvatori** [comment]
@Nate-Wilkins  @SeanFeldman @tomkerkhove  please let me know if I can proceed and merge the PR, thanks!

**Comment by Nate-Wilkins** [comment]
@tomkerkhove yeah those weren't intentional, fixed them in 7d4d01b00c0a536c9e3e59aedbdca2d769b00f46

@paolosalvatori Pretty sure this is good to go. 
Only thing that might need more attention is the appveyor file.  It also needs your chocolatey api key to push packages.

**Comment by tomkerkhove** [comment]
@Nate-Wilkins Thank you!

**Comment by SeanFeldman** [comment]
> Only thing that might need more attention is the appveyor file. It also needs your chocolatey api key to push packages.

@paolosalvatori you'll need to provide the secured  chocolatey api key. That would be the Chocolatey API Key that has been hashed using AppVeyor (can be done [here](https://ci.appveyor.com/tools/encrypt)).  The actual API key should not be stored in `AppVeyor.yml`

**Comment by paolosalvatori** [comment]
@SeanFeldman I sent you the my chocolatey key encrypted with the AppVeyor Encrypt Data tool. Let me know if you need anything else. @Nate-Wilkins @tomkerkhove

**Comment by SeanFeldman** [comment]
@Nate-Wilkins could you please apply this encrypted chocolatey key to your PR to verify build is now green? Thanks.

```
68Fr+kbT0ilhb84AyU/kKVWJw74urnreRDCpZ2L89v0cJeuEGcOchKGIFsy4Hp+V
```

File: src/ServiceBusExplorer/ServiceBusExplorer.nuspec [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    <version>$version$</version>
+    <authors>paolosalvator</authors>
+    <owners>paolosalvator</owners>
+    <summary>The Service Bus Explorer allows users to connect to a Service Bus namespace and administer messaging entities in an easy manner. The tool provides advanced features like import/export functionality or the ability to test topic, queues, subscriptions, relay services, notification hubs and events hubs.</summary>
+    <description>The Service Bus Explorer allows users to connect to a Service Bus namespace and administer messaging entities in an easy manner. The tool provides advanced features like import/export functionality or the ability to test topic, queues, subscriptions, relay services, notification hubs and events hubs.</description>
+    <projectUrl>https://github.com/paolosalvatori/ServiceBusExplorer</projectUrl>
+    <projectSourceUrl>https://github.com/paolosalvatori/ServiceBusExplorer</projectSourceUrl>
+    <packageSourceUrl>https://github.com/paolosalvatori/ServiceBusExplorer</packageSourceUrl>
+    <tags>servicebusexplorer azure service bus explorer</tags>
+    <copyright>Copyright 2017 Microsoft</copyright>
```
Comment (SeanFeldman) [inline src/ServiceBusExplorer/ServiceBusExplorer.nuspec]: @paolosalvatori is this accurate?

File: appveyor.yml [hunk 7]
```diff
@@ -66,10 +67,14 @@ deploy:
   - provider: GitHub
     auth_token:
       secure: SDw1gs4noXA0eUpBoBcbT2XNYwf0Bg7gHR/TpP2Y/gQS/QYST7toE0oRzoKV/2U6 # encrypted token from GitHub
+    choco_token:
+      secure:
```
Comment (SeanFeldman) [inline appveyor.yml]: You can use encrypted key provided by Paolo [here](https://github.com/paolosalvatori/ServiceBusExplorer/pull/140#issuecomment-315788461)

File: appveyor.yml [hunk 8]
```diff
@@ -39,6 +39,7 @@ build:
 
 after_build:
   7z a ServiceBusExplorer-%GitVersion_MajorMinorPatch%.zip %APPVEYOR_BUILD_FOLDER%\src\ServiceBusExplorer\bin\Release\*.*
+  nuget pack %APPVEYOR_BUILD_FOLDER%\src\ServiceBusExplorer\ServiceBusExplorer.nuspec -Properties="version=%GitVersion_MajorMinorPatch%"
```
Comment (SeanFeldman) [inline appveyor.yml]: 👍

File: src/ServiceBusExplorer/ServiceBusExplorer.nuspec (hunk 6 above)
Comment (Nate-Wilkins) [inline src/ServiceBusExplorer/ServiceBusExplorer.nuspec]: Guessing that's very wrong er what should the copyright be?

Comment (SeanFeldman) [inline src/ServiceBusExplorer/ServiceBusExplorer.nuspec]: You can't go wrong by making @paolosalvatori the one :)

Comment (Nate-Wilkins) [inline src/ServiceBusExplorer/ServiceBusExplorer.nuspec]: `Copyright 2017 The One` I like it

**Comment by Nate-Wilkins** [comment]
I don't think tags are supposed to be prefixed with a `#` Otherwise let's do this! 👍

**Comment by SeanFeldman** [comment]
@paolosalvatori @Nate-Wilkins the build is failing. Do not merge until it's green.
I'm travelling and can't spend time on it, but if you get stack, I'll try to give a hand.

**Comment by paolosalvatori** [comment]
Ok, just remove the # from the tags. The rest should be ok.

**Comment by tomkerkhove** [comment]
I opened a PR on your repo @Nate-Wilkins to fix the `#` since I can't fix it on your branch directly.

**Comment by tomkerkhove** [comment]
Also changed the way we pass the choco key based on docs: https://github.com/chocolatey/choco/wiki/CommandsPush

**Comment by SeanFeldman** [comment]
:x: [Build ServiceBusExplorer 4.0.100-PullRequest0140.13.build.269 failed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/build/4.0.100-PullRequest0140.13.build.269) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/07b8ad445a by @Nate-Wilkins)

**Comment by SeanFeldman** [comment]
:x: [Build ServiceBusExplorer 4.0.100-PullRequest0140.14.build.270 failed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/build/4.0.100-PullRequest0140.14.build.270) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/b44c797f78 by @Nate-Wilkins)

**Comment by SeanFeldman** [comment]
:x: [Build ServiceBusExplorer 4.0.100-PullRequest0140.15.build.271 failed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/build/4.0.100-PullRequest0140.15.build.271) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/51da06682d by @Nate-Wilkins)

**Comment by SeanFeldman** [comment]
:x: [Build ServiceBusExplorer 4.0.100-PullRequest0140.16.build.272 failed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/build/4.0.100-PullRequest0140.16.build.272) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/74fb532dea by @Nate-Wilkins)

**Comment by SeanFeldman** [comment]
:x: [Build ServiceBusExplorer 4.0.100-PullRequest0140.17.build.273 failed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/build/4.0.100-PullRequest0140.17.build.273) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/cd00a73fdc by @Nate-Wilkins)

**Comment by SeanFeldman** [comment]
:x: [Build ServiceBusExplorer 4.0.100-PullRequest0140.18.build.274 failed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/build/4.0.100-PullRequest0140.18.build.274) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/b0c5353a03 by @Nate-Wilkins)

**Comment by SeanFeldman** [comment]
:x: [Build ServiceBusExplorer 4.0.100-PullRequest0140.19.build.275 failed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/build/4.0.100-PullRequest0140.19.build.275) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/f40cf80f11 by @Nate-Wilkins)

**Comment by SeanFeldman** [comment]
:x: [Build ServiceBusExplorer 4.0.100-PullRequest0140.20.build.276 failed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/build/4.0.100-PullRequest0140.20.build.276) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/f40cf80f11 by @Nate-Wilkins)

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBusExplorer 4.0.100-PullRequest0140.21.build.277 completed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/build/4.0.100-PullRequest0140.21.build.277) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/36d4991452 by @Nate-Wilkins)

**Comment by Nate-Wilkins** [comment]
Wow that wasn't fun... I removed the ability for PRs to push up chocolatey packages so this should push up the latest chocolatey package when this gets merged into `master`

**Comment by paolosalvatori** [comment]
Can I safely merge the PR? I lost the count :)

**Comment by Nate-Wilkins** [comment]
Yes we're good to go! 😇

**Comment by SeanFeldman** [comment]
Thank you @Nate-Wilkins. Could you please squash the commits or @paolosalvatori you could squash merge. I can review tonight if you can't.

**Comment by Nate-Wilkins** [comment]
Yeah that's a great idea, - nobody needs to see those commits anyhow 👍

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBusExplorer 4.0.100-PullRequest0140.2.build.278 completed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/build/4.0.100-PullRequest0140.2.build.278) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/dcc05c451d by @Nate-Wilkins)

**Comment by SeanFeldman** [comment]
Thank you @Nate-Wilkins! SBE should be available on Chocholatey as of this release 🙂

**Comment by paolosalvatori** [comment]
Thanks @Nate-Wilkins :) Thanks everyone on this thread :)

**Pull request merged**